hnn_core/viz.py

```diff
@@ -64,21 +77,69 @@ def plot_hist_input(net, ax=None, show=True):
         The matplotlib figure handle.
     """
     import matplotlib.pyplot as plt
-    spikes = np.array(sum(net.spikes.times, []))
-    gids = np.array(sum(net.spikes.gids, []))
-    valid_gids = np.r_[[v for (k, v) in net.gid_dict.items()
-                        if k.startswith('evprox')]]
-    mask_evprox = np.in1d(gids, valid_gids)
-    valid_gids = np.r_[[v for (k, v) in net.gid_dict.items()
-                        if k.startswith('evdist')]]
-    mask_evdist = np.in1d(gids, valid_gids)
-    bins = np.linspace(0, net.params['tstop'], 50)
+    spike_times = np.array(sum(spikes._times, []))
+    spike_types_data = np.array(sum(spikes._types, []))
+
+    unique_types = np.unique(spike_types_data)
+    spike_types_mask = {s_type: np.in1d(spike_types_data, s_type)
+                        for s_type in unique_types}
+    cell_types = ['L5_pyramidal', 'L5_basket', 'L2_pyramidal', 'L2_basket']
+    input_types = np.setdiff1d(unique_types, cell_types)
```

Comment:
didn't we agree that we'll make this more general for any spike type. Did you check that it works for e.g., if you provide `'L5_pyramidal'`?

Reply:
Might be a miscommunication, this was based on our discussion [here](https://github.com/jonescompneurolab/hnn-core/pull/157#discussion_r479691175). The None condition checks if there are any input spikes and plots them by default. Otherwise it plots spikes from all cells. Defining the type works the same for everything though:
![image](https://user-images.githubusercontent.com/55253912/91663670-50e09900-eab8-11ea-8273-c9817c80e392.png)

Were you thinking of some other behavior?

Reply:
This looks nice to me. Is the part about "Otherwise it plots spikes from all cells." documented in the docstring?

Also maybe we can update the example with this plot? to show that we can do it for any cell type

and also mention that this histogram is accumulated across trials, right?